### PR TITLE
Add xcarchive build support

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR [version update](https://semver.org/)

### Context

Step can't create an xcarchive at the moment, only an app bundle. This is not sufficient when trying to create a deploy workflow that uploads an xcarchive/IPA to the store or somewhere else.

### Changes

- New step input: `ios_output_type` with two values `app` (builds app bundle like before) and `archive` (builds xcarchive). The default is `app`, so it's not a breaking change.
- New outputs: `$BITRISE_XCARCHIVE_PATH` and `$BITRISE_XCARCHIVE_ZIP_PATH`
- The xcarchive zip is also placed in `$BITRISE_DEPLOY_DIR`
- The default value of the `ios_additional_params` input is no longer `--release --no-code-signing` because the `flutter build ipa` command doesn't understand the `--no-code-signing` flag (it doesn't make sense). The `--no-code-signing` flag is now added dynamically when `ios_output_type` is `app` to preserve backwards compatibility.
- Internal refactors to simplify the Flutter CLI command creation
- Some cleanup in `e2e/bitrise.yml` (see comments)

### Decisions

No breaking change in step behavior, so the default step inputs should produce the same outputs.

### CODE REVIEW

File: buildspec.go [hunk 1]
```diff
@@ -55,10 +57,6 @@ func (spec buildSpecification) artifactPaths(outputPathPatterns []string, isDir
 }
 
 func (spec buildSpecification) exportIOSApp(artifacts []string, deployDir string) error {
-	if len(artifacts) < 1 {
```
Comment (ofalvai) [inline buildspec.go:58]: This is not needed, the callsite in `main.go` checks the same before calling this.

File: e2e/bitrise.yml [hunk 2]
```diff
@@ -268,25 +299,7 @@ workflows:
   _setup_test:
     steps:
     - script:
-        run_if: $.IsCI
-        inputs:
-        - content: |-
-            #!/usr/bin/env bash
-
-            yes | sdkmanager --install "build-tools;28.0.3"
```
Comment (ofalvai) [inline e2e/bitrise.yml:276]: It was a workaround for the Xamarin stack, no longer needed

File: e2e/bitrise.yml [hunk 3]
```diff
@@ -268,25 +299,7 @@ workflows:
...
-
-            set -ex
-            if [[ "$OSTYPE" == "linux-gnu"* ]]; then
```
Comment (ofalvai) [inline e2e/bitrise.yml:279]: Java 11 is the default now, not needed either

File: main.go [hunk 4]
```diff
@@ -17,37 +17,40 @@ import (
...
-	AndroidAdditionalParams string              `env:"android_additional_params"`
-	Platform                string              `env:"platform,opt[both,ios,android]"`
-	IOSExportPattern        string              `env:"ios_output_pattern,required"`
-	AndroidOutputType       AndroidArtifactType `env:"android_output_type,opt[apk,appbundle]"`
-	AndroidExportPattern    string              `env:"android_output_pattern,required"`
-	IOSCodesignIdentity     string              `env:"ios_codesign_identity"`
-	ProjectLocation         string              `env:"project_location,dir"`
-	DebugMode               bool                `env:"is_debug_mode,opt[true,false]"`
-	CacheLevel              string              `env:"cache_level,opt[all,none]"`
+	ProjectLocation       string `env:"project_location,dir"`
```
Comment (ofalvai) [inline main.go:37]: Reordered and grouped inputs, plus the one new iOS input

**Review by shams-ahmed — APPROVED** [review]

File: buildspec.go [hunk 5]
```diff
@@ -173,9 +194,23 @@ func (spec buildSpecification) build(params string) error {
...
+	}
+
+	if spec.platformOutputType == OutputTypeIOSApp {
+		paramSlice = append(paramSlice, "--no-codesign")
+	}
+
+	buildCmd := command.New("flutter", append([]string{"build", platformCmd}, paramSlice...)...).SetStdout(os.Stdout)
 
-	if spec.platformCmdFlag == "ios" {
+	if spec.platformOutputType == OutputTypeIOSApp {
```
Comment (lpusok) [inline buildspec.go:213]: Can these specially handled cases happen for archive too?

Comment (ofalvai) [reply buildspec.go:213]: I did not encounter this in my testing, but I'll add the "archive" case here too just in case.

**Review by lpusok — APPROVED** [review]

**Pull request merged**